File: backend/base/src/main/java/de/eshg/base/address/persistence/entity/Address.java
/*
 * Copyright 2025 cronn GmbH
 * SPDX-License-Identifier: Apache-2.0
 */

package de.eshg.base.address.persistence.entity;

import de.eshg.lib.common.CountryCode;

public interface Address {

  Long getId();

  String getPostalCode();

  void setPostalCode(String postalCode);

  String getCity();

  void setCity(String city);

  CountryCode getCountry();

  void setCountry(CountryCode country);

  String getDifferentName();

  void setDifferentName(String differentName);
}

File: apps/citizen-portal/src/lib/shared/formatters/address.ts
/**
 * Copyright 2025 cronn GmbH
 * SPDX-License-Identifier: Apache-2.0
 */

import { ApiGetDepartmentInfoResponse } from "@eshg/base-api";
import {
  formatList,
  formatPostalCodeAndCity,
  formatStreetAndHouseNumber,
} from "@eshg/lib-portal";

export function formatDepartmentAddress(
  department: ApiGetDepartmentInfoResponse,
) {
  return formatList(
    [
      department.name,
      formatStreetAndHouseNumber(department),
      formatPostalCodeAndCity(department),
    ],
    ", ",
  );
}
